src/components/ui/Image/useSources.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react'
+import type { SourceHTMLAttributes } from 'react'
+
+import { urlBuilder } from './thumborUrlBuilder'
+import type { ThumborOptions } from './thumborUrlBuilder'
+
+export interface SourceProps extends SourceHTMLAttributes<HTMLSourceElement> {
+  src: string
+  width: number
+  height: number
+  media: string
+}
+
+export interface SourceOptions extends SourceProps {
+  options?: ThumborOptions
+}
+
+const FACTORS = [1, 2, 3]
```

Comment:
Are these multiplying factors always these numbers?

Reply:
These FACTORS were based on the previously implemented `useImage`. 

 These are the most common values based on the device's density. You can type `devicePixelRatio` on `devtools` console to see the value for each selected device.